src/components/navs/AppNav/AppNavClaimBtn.vue

```diff
@@ -0,0 +1,201 @@
+          <div class="text-lg font-bold">
+            {{ fNum(currentRewards, 'token') }} BAL
+          </div>
+          <div>
+            {{
+              currentRewardsInUSD != null
+                ? fNum(currentRewardsInUSD, 'usd')
+                : '-'
+            }}
+          </div>
+        </div>
+      </div>
+    </div>
+  </BalPopover>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref } from 'vue';
+import { useStore } from 'vuex';
+import { differenceInSeconds } from 'date-fns';
+import { useIntervalFn } from '@vueuse/core';
+
+import useNumbers from '@/composables/useNumbers';
+import useWeb3 from '@/composables/useWeb3';
+import useAuth from '@/composables/useAuth';
+import useNotify from '@/composables/useNotify';
+import useUserClaimsQuery from '@/composables/queries/useUserClaimsQuery';
+
+import { getOriginalAddress } from '@/services/coingecko';
+
+import { TOKENS } from '@/constants/tokens';
+import { bnum } from '@/lib/utils';
+import { claimRewards } from '@/services/claim';
+
+export default defineComponent({
+  name: 'AppNavClaimBtn',
+
+  setup() {
+    // DATA
+
+    const isClaiming = ref(false);
+    const rewardsEstimateSinceTimestamp = ref('0');
+
+    // COMPOSABLES
+    const store = useStore();
+    const userClaimsQuery = useUserClaimsQuery();
+    const { fNum } = useNumbers();
+    const { appNetwork, account } = useWeb3();
+    const { txListener } = useNotify();
+    const auth = useAuth();
+
+    const balPrice = computed(
+      () =>
+        store.state.market.prices[
+          getOriginalAddress(appNetwork.id, TOKENS.AddressMap.BAL)
+        ]?.price
+    );
+
+    // COMPUTED
+    const userClaims = computed(() =>
+      userClaimsQuery.isSuccess.value ? userClaimsQuery.data?.value : null
+    );
+
+    const availableToClaimInUSD = computed(() =>
+      balPrice.value != null && userClaims.value != null
+        ? bnum(userClaims.value?.availableToClaim)
+            .times(balPrice.value)
+            .toString()
+        : null
+    );
+
+    const currentRewards = computed(() =>
+      userClaims.value != null &&
+      userClaims.value.currentRewardsEstimate != null
+        ? bnum(userClaims.value.currentRewardsEstimate.rewards)
+            .plus(rewardsEstimateSinceTimestamp.value)
+            .toString()
+        : null
+    );
+
+    const currentRewardsInUSD = computed(() =>
+      balPrice.value != null && currentRewards.value != null
+        ? bnum(currentRewards.value)
+            .times(balPrice.value)
+            .toString()
+        : null
+    );
+
+    const totalRewards = computed(() =>
+      userClaims.value != null
+        ? bnum(userClaims.value.totalRewards)
+            .plus(rewardsEstimateSinceTimestamp.value)
+            .toString()
+        : null
+    );
+
+    useIntervalFn(async () => {
+      if (userClaims.value != null && userClaims.value.currentRewardsEstimate) {
+        const diffInSeconds = differenceInSeconds(
```

Comment:
How about using `differenceInMilliseconds` instead and a higher frequency (eg updates every 30 milliseconds) for a more fluid look? Or would that be too memory intensive?

Reply:
I actually tried it - but it didn't look great. I think I can definitely improve on this... but perhaps in a later PR?